prog/vnet/rekey_nic.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+class Prog::Vnet::RekeyNic < Prog::Base
+  subject_is :nic
+
+  def start
+    nic.dst_ipsec_tunnels.each do |tunnel|
```

Comment:
Do you have sketch of a proof why this traversal is robust to additions/subtractions/modifications concurrent with the enumeration of the collection?